extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxCurrentContextFactory.java

```diff
@@ -41,20 +56,47 @@ public <T extends InjectableContext.ContextState> CurrentContext<T> create(Class
         return new VertxCurrentContext<>(key);
     }
 
-    /**
-     *
-     * @return an unmodifiable list of used keys
-     */
-    public List<String> keys() {
-        return unmodifiableKeys;
+    public ContextInternal duplicateContext(ContextInternal vertxContext) {
+        List<String> keys = this.keys;
+        // TODO optimize it to do it right!
+        ConcurrentMap<Object, Object> local = vertxContext.localContextData();
+        if (containsScopeKey(keys, local)) {
```

Comment:
This would return `false` if `local` only contains the `REQUEST_SCOPED_LOCAL_KEY`, or?

The list of keys does not contain the special key for `@RequestScoped`.

Reply:
I have used the atomic boolean remove from the other context the request scoped key, if any

Reply:
My point is that `containsScopeKey(keys, local)` only checks the list of keys. So if no string key is used (`REQUEST_SCOPED_LOCAL_KEY` is not added to the list) then this method will never create a new duplicated context...